.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,39 @@
+name: e2e-mock-provider-tests
```

Comment:
Could we use underscore in the names for workflow consistency in our project? 

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/fae105dffd45178501cbec18328c55a02d8257dc/.github/workflows/chart.yaml#L1

Reply:
done